### PR TITLE
Remove rcf jar and fix zip fetching for AD and JS

### LINKED ISSUE #433: [BUG] Plugin contains checked in JARs with unclear provenance
**Describe the bug**

The lib/randomcutforest-parkservices-2.0.1.jar and randomcutforest-core-2.0.1.jar JARs have unclear provenance. These need to be external dependencies. Code comes from https://github.com/aws/random-cut-forest-by-aws

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
1. Removed RCF jars and updated to fetch RCF 3.0-rc2 from maven
2. Added zip and folder deletion before fetching zips for bwc and integCluster so there isn't multiple zips 
3. added bwc back to CI
4. changed `V1JsonToV2StateConverter` usage `V1JsonToV3StateConverter`
5. Added testDeserialize based on 1.3 rcf model
 
### Issues Resolved
resolves #433 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java [hunk 1]
```diff
@@ -154,7 +157,7 @@
     private GenericObjectPool<LinkedBuffer> serializeRCFBufferPool;
     private RandomCutForestMapper mapper;
     private ThresholdedRandomCutForestMapper trcfMapper;
-    private V1JsonToV2StateConverter converter;
+    private V1JsonToV3StateConverter converter;
```
Comment (kaituo) [inline src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java:160]: There is also a V2StateToV3ForestConverter. We need to convert from v1 to v3, v2 to v3. If v2 and v3 are incompatible, we need to add a field modelV3 in checkpoint.

File: src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java [hunk 2]
```diff
@@ -1000,4 +1003,14 @@ public void testFromEntityModelCheckpointWithEntity() throws Exception {
         }
         return point;
     }
+    // .parseReader(new FileReader(new File(getClass().getResource(labalFileName).toURI()), Charset.defaultCharset()))
+
+    public void testDeserializeRCFModel() throws Exception {
+        // Model in file 1_3_0_rcf_model.json not passed initialization yet
```
Comment (kaituo) [inline src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java:1009]: So it means the model is empty? If yes, have you tested model fully initialized?

File: build.gradle [hunk 3]
```diff
@@ -590,6 +607,9 @@ dependencies {
     compileOnly "org.opensearch:opensearch-job-scheduler-spi:${job_scheduler_version}"
     implementation "org.opensearch:common-utils:${common_utils_version}"
     implementation "org.opensearch.client:opensearch-rest-client:${opensearch_version}"
+    implementation 'software.amazon.randomcutforest:randomcutforest-parkservices:3.0-rc2'
```
Comment (kaituo) [inline build.gradle:610]: Could you list the tests you have done to upgrade to rc2? What I can think of:
* real time single stream and HCAD (including create/start detector, see detector emit results; then stop cluster and restart it to see if results continue to show)
* historical single stream and HCAD
* backward compatible tests: v1 to v3 model, v2 to v3 model: v1 model in checkpoint should not be overridden, and we add v2 and v3 model in checkpoint if necessary.
* Does memory size formula still hold (Check MemoryTracker)?

File: src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java [hunk 4]
```diff
@@ -1000,4 +1003,14 @@ public void testFromEntityModelCheckpointWithEntity() throws Exception {
         }
         return point;
     }
+    // .parseReader(new FileReader(new File(getClass().getResource(labalFileName).toURI()), Charset.defaultCharset()))
```
Comment (kaituo) [inline src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java:1006]: unused code

File: build.gradle (hunk 3 above)
Comment (amitgalitz) [reply build.gradle:610]: I tested single stream real time and historical, I;ll do the same for HCAD right now. BWC test were the test added based on Yaliang's code to check deserialize model from 1.3. And converting test to v1 to v3, I didn't add v2 to v3.

File: src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java (hunk 4 above)
Comment (amitgalitz) [reply src/test/java/org/opensearch/ad/ml/CheckpointDaoTests.java:1006]: will remove

File: build.gradle (hunk 3 above)
Comment (kaituo) [reply build.gradle:610]: The unit tests for bwc is not enough. It just checks the conversion has no exception. We will do some manual testing suggested above.

Comment (amitgalitz) [reply build.gradle:610]: I manually tested that after starting both single and HCAD (both real time and historical) with current RCF and getting results, I can switch to RCF 3.0-rc2 and there is no issues, detectors continue to run the same

Comment (amitgalitz) [reply build.gradle:610]: After letting the detector run for longer I do get one issue, `Failed to generate checkpoint, java.lang.IllegalStateException: There is discepancy in indices`

**Comment by amitgalitz** [comment]
Closing in favor of new PR that fetches RCF 3.0-rc1 instead of 3.0-rc2. There are some compatibility issue that need further investigation for AD transition to rc2.

**Pull request closed without merging**